pkg/multiclustermanager/routes.go

```diff
@@ -111,9 +110,21 @@ func router(ctx context.Context, localClusterEnabled bool, tunnelAuthorizer *mcm
 	authed.PathPrefix("/v3/token").Handler(tokenAPI)
 	authed.PathPrefix("/v3").Handler(managementAPI)
 
+	// Metrics authenticated route
+	metricsAuthed := mux.NewRouter()
+	metricsAuthed.UseEncodedPath()
+	impersonatingAuth := auth.ToMiddleware(requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager)))
```

Comment:
can we define this above and use this in https://github.com/rancher/rancher/blob/978e165156362d98a323452049fb04c7fec64f5d/pkg/multiclustermanager/routes.go#L96 as well?

Based on the fact that the logic for [`Chain`](https://github.com/rancher/steve/blob/e7119828b8f9a9ca2cb2620ba650bd0508ca463e/pkg/auth/filter.go#L47-L51) emits a brand new function, we should be able to reuse the same function for impersonating auth in both places rather than defining it twice.